libs/constrained-generators/src/Constrained/Spec/Map.hs

```diff
@@ -56,13 +56,59 @@ data MapSpec fn k v = MapSpec
 defaultMapSpec :: Ord k => MapSpec fn k v
 defaultMapSpec = MapSpec Nothing mempty mempty TrueSpec TrueSpec NoFold
 
-deriving instance
+instance
+  ( Arbitrary k
+  , Arbitrary v
+  , Arbitrary (TypeSpec fn k)
+  , Arbitrary (TypeSpec fn v)
+  , Ord k
+  , HasSpec fn k
+  , Foldy fn v
+  ) =>
+  Arbitrary (MapSpec fn k v)
+  where
+  arbitrary =
+    MapSpec
+      <$> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+      <*> frequency [(5, pure NoFold), (1, arbitrary)]
+
+instance Arbitrary (FoldSpec fn (Map k v)) where
+  arbitrary = pure NoFold
```

Comment:
I guess you'll want to make this more interesting in the future? Maybe add a `TODO` so you don't forget.

Reply:
Right now it works well enough, the problem is that it causes unnecessarily many calls to `genFromSpec_` to fail.

Reply:
But I've added a TODO.

Reply:
Yeah, I just figured you wanted to add something more here, since just above this you have 
```haskell
frequency [(5, pure NoFold), (1, arbitrary)]
```
which is as good as having `arbitrary` there instead

Reply:
Yeah, this was actually an attempt to solve this exact problem of unsat distributions. If you have too many fold constraints you end up with most maps being completely unsolveable. That's why this tweaks the underlying distribution to generate meaningful constraints here more rarely.